src/options.rs

```diff
@@ -235,16 +234,8 @@ fn conn_params(tmp: &TmpOptions) -> anyhow::Result<Builder> {
         user.map(|user| conn_params.user(user));
         database.map(|database| conn_params.database(database));
     } else {
-        let user = user.unwrap_or_else(|| {
-            if admin {
-                String::from("edgedb")
-            } else {
-                whoami::username()
-            }
-        });
-        conn_params.user(user.clone());
-        conn_params.database(database.as_ref().map(|x| &x[..])
-                             .unwrap_or(&user));
+        user.as_ref().map(|user| conn_params.user(user));
+        database.as_ref().map(|db| conn_params.database(db));
```

Comment:
Just for my education -- the code in the `if` branch appears almost identical, except there it's `user.map(...)` and here it's `user.as_ref().map(...)`. What's the difference?

Reply:
`as_ref()` converts `Option<String>` into `Option<&String>`. Without `as_ref()` the method `user.map(|user:String| { })` gets the string itself, not the reference. So it effectively moves out `user` variable and it can't be used later in code. In that if above it looks like we don't use `user` anymore. (we could write `as_ref()` above too, but not vice versa).